zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/IndexMappingProperty.java

```diff
@@ -7,32 +7,40 @@
  */
 package io.camunda.exporter.schema;
 
+import static java.util.Map.entry;
+
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
+import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import org.apache.commons.io.IOUtils;
 
 public record IndexMappingProperty(String name, Object typeDefinition) {
 
-  public static InputStream toPropertiesJson(
-      final Set<IndexMappingProperty> properties, final ObjectMapper mapper) {
-    final var propertiesAsMap =
-        properties.stream()
-            .collect(
-                Collectors.toMap(IndexMappingProperty::name, IndexMappingProperty::typeDefinition));
-    final var propertiesBlock = new HashMap<>();
-    propertiesBlock.put("properties", propertiesAsMap);
-    try {
-      return IOUtils.toInputStream(
-          mapper.writeValueAsString(propertiesBlock), StandardCharsets.UTF_8);
-    } catch (final JsonProcessingException e) {
-      throw new RuntimeException(e);
-    }
+  public static <T> Map<String, T> toPropertiesMap(
+      final Set<IndexMappingProperty> properties,
+      final ObjectMapper mapper,
+      final Function<InputStream, T> typeDeserializer) {
+
+    return properties.stream()
+        .map(
+            prop -> {
+              try {
+                final var typeJson =
```

Comment:
❓ where did these variable names (typeJson, type) come from? what do we mean by type here? feels confusing to me

Reply:
This is actually out of date, I made the serialisation generic, it was changed here https://github.com/camunda/camunda/pull/23393/files, It would be easier to review that separately, in the mean time I will rebase it in to here.

Reply:
This has been refactored into `IndexMappingProperty.toElasticsearchProperty` (with the opensearch implementation to be created in the next PR) which converts an IndexMappingProperty back into a elasticsearch index mapping property. 

The new variable names and function names should be a lot clearer